R/PipeOpFDAInterpol.R

```diff
@@ -0,0 +1,140 @@
+#' * `left` :: `numeric()` \cr
+#'   The left boundary of the window.
+#'   The window is specified such that the all values >=left and <=right are kept for the computations.
+#' * `right` :: `numeric()` \cr
+#'   The right boundary of the window.
+#'
+#' @export
+#' @examples
+#' library(mlr3pipelines)
+#' task = tsk("fuel")
+#' pop = po("fda.interpol")
+#' task_interpol = pop$train(list(task))
+PipeOpFDAInterpol = R6Class("PipeOpFDAInterpol",
+  inherit = mlr3pipelines::PipeOpTaskPreprocSimple,
+  public = list(
+    #' @description Initializes a new instance of this Class.
+    #' @param id (`character(1)`)\cr
+    #'   Identifier of resulting object, default `"fda.interpol"`.
+    #' @param param_vals (named `list`)\cr
+    #'   List of hyperparameter settings, overwriting the hyperparameter settings that would
+    #'   otherwise be set during construction. Default `list()`.
+    initialize = function(id = "fda.interpol", param_vals = list()) {
+      param_set = ps(
+        grid = p_uty(tags = c("train", "predict", "required"), custom_check = crate(function(x) {
+          if (test_string(x)) {
+            return(check_choice(x, choices = c("union", "intersect", "minmax")))
+          }
+          if (test_numeric(x, any.missing = FALSE, min.len = 1)) {
+            return(TRUE)
+          }
+          "Must be either a string or numeric vector."
+        })),
+        method = p_fct(
+          c("linear", "spline", "fill_extend", "locf", "nocb"), default = "linear", tags = c("train", "predict")
+        ),
+        left = p_dbl(tags = c("train", "predict")),
+        right = p_dbl(tags = c("train", "predict"))
+      )
+      param_set$set_values(grid = "union")
+
+      super$initialize(
+        id = id,
+        param_set = param_set,
+        param_vals = param_vals,
+        packages = c("mlr3fda", "mlr3pipelines"),
+        feature_types = c("tfd_reg", "tfd_irreg")
+      )
+    }
+  ),
+  private = list(
+    .transform_dt = function(dt, levels) {
+      pars = self$param_set$get_values()
+      grid = pars$grid
+      method = pars$method
+      left = pars$left
+      right = pars$right
+      if (!is.null(left) && !is.null(right)) {
+        assert_count(grid)
+        assert_true(left <= right)
+      }
+      method = method %??% "linear"
+      evaluator = sprintf("tf_approx_%s", method)
+      map_dtc(dt, function(x) interpolate_col(x, grid, evaluator, left, right))
+    }
+  )
+)
+
+interpolate_col = function(x, grid, evaluator, left, right) {
+  if (is.numeric(grid)) {
+    if (length(grid) > 1L && is.null(left) && is.null(right)) {
+      arg = unlist(tf::tf_arg(x))
+      if (max(grid) > max(arg) || min(grid) < min(arg)) {
+        stopf("The grid must be within the range of the argument points.")
+      }
+      arg = grid
+    } else {
+      arg = seq(left, right, length.out = grid)
+    }
+  } else if (tf::is_reg(x)) {
```

Comment:
what about "minmax", should this not have an effect for regular?

Reply:
no why?